python/sglang/srt/managers/scheduler.py

```diff
@@ -293,6 +296,9 @@ def __init__(
             _,
             _,
         ) = self.tp_worker.get_worker_info()
+        if self.device == "hpu":
+            # scheduler will run on cpu if hpu is used
+            self.device = "cpu"
```

Comment:
Better to add  `get_scheduler_device` in `srt/utils.py`

Reply:
We can use `use_pure_cpu_scheduler` here if we implement it

Reply:
Fixed. I added a `get_scheduler_device` function